proto/src/protobuf/erased.rs

```diff
@@ -0,0 +1,40 @@
+use alloc::boxed::Box;
+use core::convert::{Into as CoreInto, TryFrom as CoreTryFrom};
+
+mod sealed {
+    use super::*;
+
+    pub trait SealedInto<T: ?Sized> {}
+    impl<T, U: Clone + CoreInto<T>> SealedInto<T> for U {}
+    pub trait SealedTryFrom<T> {}
+    impl<T, U: CoreTryFrom<T>> SealedTryFrom<T> for U {}
+}
+
+pub trait Into<T: ?Sized>: sealed::SealedInto<T> {
+    fn into(&self) -> Box<T>;
```

Comment:
I think we should give a different name to `into()`, such as `into_ref()`. With the current version, I noticed weird behaviors in my editor. When hovering over a call to `erased::Into::into()` in one of the functions of our `Protobuf` implementation, rust-analyzer claims that the `into()` being referenced is the one in `core::convert::Into`. I even tried to explicitly write
```rust
let raw: Box<T> = self.into();
```
and rust-analyzer still thinks that I'm using `core::convert::into`. I haven't tested other language servers.

I then played around a bit in a rust playground [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ea46d1568133bf5089af0a1665b04114). I'm not sure why it's not working; perhaps it has something to do with `core::convert::Into` being in the prelude?

In any case, using another name for `into()` avoids these headaches.

Reply:
This is an interesting problem. I used `erased::Into` because that is the convention that erased_serde uses and I thought that it was a common convention, but it seems we're seeing problems because we're using it with std lib traits (in the prelude). IMHO, `into_ref()` would be a method that takes `self` by value, so a better name would perhaps be `to_boxed()`. WDYT?

Reply:
I agree that `into_ref()` suggests taking ownership. On the flip slide, for me, `to_boxed()` suggests boxing `self` (i.e. no conversion to another type), such as

```rust
fn to_boxed(&self) -> Box<Self> {
    Box::new(self.clone())
}
```

Brainstorming alternatives:
+ `into_boxed()`: still `into` suggests taking ownership of `self`
+ `clone_boxed()`: I don't like it, doesn't suggest a conversion either
+ `clone_into_boxed()`: quite wordy, but perhaps captures best the conversion we're doing?
+ `as_boxed()`: `as_*` in `std` take a `&self`, and suggest a conversion?
+ ... ?